.gitignore

```diff
@@ -1,3 +1,7 @@
 backup/*
 build/*
 input/*
+lib/*
+.vscode/*
+.cache/*
+.clang-format
```

Comment:
Please remove these - I know they're useful in development, but not in the final commit